### PR TITLE
Fix N+1 query pattern in bills API endpoint

### LINKED ISSUE #31: Fix N+1 query pattern in bills API endpoint
## Problem

The `GET /api/bills` endpoint executes a separate database query for each bill to fetch transaction counts, creating an N+1 query pattern. With 10 bills, this means 10+ queries instead of 1.

**File:** `src/app/api/bills/route.ts`

## Proposed Solution

Replace the loop of individual queries with a single query using `LEFT JOIN` and `GROUP BY` to fetch bill data and transaction counts in one pass.

## Impact

- Eliminates N+1 database queries
- Scales better as bill count grows
- Low effort fix (~30 min)

### PR DESCRIPTION
**Pull request by noahcoffey**
## Summary
- Replaces per-bill transaction matching queries (N+1) with a single batch query that fetches all monthly transactions once, then matches in JavaScript
- With 10 bills, this reduces database queries from 11 to 2
- Matching logic preserved: $1 amount tolerance, case-insensitive name matching against both `display_name` and `raw_description`
- Adds 9 tests covering payment matching edge cases

## Test plan
- [x] All 257 tests pass (`npm test`)
- [x] `npx next build` succeeds
- [x] Tests cover: exact match, raw_description fallback, amount tolerance, tolerance boundary, multiple bills, case-insensitive matching, inactive bill exclusion

Closes #31

## Summary by CodeRabbit

* **Refactor**
  * Improved bill-payment matching for faster, batched transaction lookups and centralized status determination, yielding more reliable and consistent isPaid/isDue/isOverdue indicators.
* **Tests**
  * Added comprehensive test coverage for matching logic, edge cases, and month/year boundaries to ensure correctness and prevent regressions.

### CODE REVIEW

File: src/lib/__tests__/bills-n-plus-1.test.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+       t.raw_description.toLowerCase().includes(bill.name.toLowerCase()))
+    )
+    return {
+      ...bill,
+      isPaid: !!match,
+      isDue: bill.next_due_date >= monthStart && bill.next_due_date <= monthEnd,
+      isOverdue: bill.next_due_date < today,
+    }
+  })
+}
```
[the remark this answers is not in the document]
Comment (noahcoffey) [reply src/lib/__tests__/bills-n-plus-1.test.ts]: Addressed — extracted transactionMatchesBill() and matchBillPayments() into src/lib/bill-matcher.ts. Both the route and tests now import from the same shared module.
